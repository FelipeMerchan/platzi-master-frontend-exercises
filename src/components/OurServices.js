import React from 'react'

import './styles/OurServices.css'

function OurServices() {
  return (
    <section className="OurServices">
      <div>
        <h2 className="OurServices___title">Nuestros Servicios</h2>
        <ul className="OurServices__list">
          <li>
            <a href="/" target="_blank">Consultoría Gerencial</a>
          </li>
          <li>
            <a href="/" target="_blank">Análisis Regulatorio de Políticas</a>
          </li>
          <li>
            <a href="/" target="_blank">Investigación de Mercado</a>
          </li>
          <li>
            <a href="/" target="_blank">Gestión de Proyectos</a>
          </li>
          <li>
            <a href="/" target="_blank">Regulación de Soporte</a>
          </li>
          <li>
            <a href="/" target="_blank">Análisis de Riesgos</a>
          </li>
          <li>
            <a href="/" target="_blank">Formación Gerencial</a>
          </li>
        </ul>
      </div>
      <div>
        <a className="button" href="/" target="_blank">Leer más</a>
      </div>
    </section>
  )
}

export default OurServices