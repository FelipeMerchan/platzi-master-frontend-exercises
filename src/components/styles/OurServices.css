.OurServices {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 30%;
  font-family: var(--font-3);
}

.OurServices___title {
  margin-bottom: var(--space);
  padding-bottom: var(--space);
  border-bottom: 1px solid #dfdfdf;
  color: #48504d;
}

.OurServices__list li {
  margin: var(--space) 0;
  list-style-type: none;
}

.OurServices__list a {
  color: var(--actionColor);
}

.OurServices__list a::before {
  content: '';
  display: inline-block;
  width: 6px;
  height: 10px;
  margin-right: var(--space);
  background-image: url('../../images/bullet-points.svg');
  background-size: contain;
  background-repeat: no-repeat;
}