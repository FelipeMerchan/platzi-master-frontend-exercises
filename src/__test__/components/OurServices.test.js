import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import OurServices from '../../components/OurServices'

describe('Probar el componente <OurServices />', () => {
  const ourServices = shallow(<OurServices />)
  test('Render del componente OurServices', () => {
    expect(ourServices.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(ourServices.find('.OurServices___title').text()).toEqual('Nuestros Servicios')
  })
  test('Render de los 7 servicios que ofrecemos', () => {
    expect(ourServices.find('.OurServices__list').children()).toHaveLength(7);
  });
})

describe('OurServices Snapshot', () => {
  test('Comprobar la UI del componente OurServices', () => {
    const ourServices = create(<OurServices />)
    expect(ourServices.toJSON()).toMatchSnapshot()
  })
})