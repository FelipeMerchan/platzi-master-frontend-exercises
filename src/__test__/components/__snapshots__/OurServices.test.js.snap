// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurServices Snapshot Comprobar la UI del componente OurServices 1`] = `
<section
  className="OurServices"
>
  <div>
    <h2
      className="OurServices___title"
    >
      Nuestros Servicios
    </h2>
    <ul
      className="OurServices__list"
    >
      <li>
        <a
          href="/"
          target="_blank"
        >
          Consultoría Gerencial
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Análisis Regulatorio de Políticas
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Investigación de Mercado
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Gestión de Proyectos
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Regulación de Soporte
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Análisis de Riesgos
        </a>
      </li>
      <li>
        <a
          href="/"
          target="_blank"
        >
          Formación Gerencial
        </a>
      </li>
    </ul>
  </div>
  <div>
    <a
      className="button"
      href="/"
      target="_blank"
    >
      Leer más
    </a>
  </div>
</section>
`;
